pkg/server/share_manager.go

```diff
@@ -105,6 +123,18 @@ func (m *ShareManager) Run() error {
 			}
 
 			m.logger.Info("Starting nfs server, volume is ready for export")
+
+			if err := m.getLease(); err != nil {
+				m.logger.WithError(err).Warn("Failed to get lease - no lease renewal will be done")
+			} else {
+				m.logger.Info("Found lease")
+				if err = m.takeLease(); err != nil {
+					m.logger.WithError(err).Warn("Failed to take lease - no lease renewal will be done.")
+				} else {
+					go m.runLeaseRenew()
```

Comment:
Don't we launch this goroutine (and take the rest of these steps) every five seconds? That seems unintended.

Reply:
Need the goroutine?

Reply:
The `select` default path does a lot of setup, and then calls Run on the nfsServer object, which waits on the ganesha process.  It will block until that process exits.  That is one way out of the `select` - the other is the context.Done channel read.
If another instance of the goroutine were launched every five seconds, the same would be true of the runHealthCheck goroutine, whose implementation I just echoed.  But it is absolutely necessary to renew the lease consistently, or share_manager_controller will restart the pod, just as it is important to run the health check.

Reply:
Got it. The part I was missing was the blocking call that ensures we don't repeatedly execute the loop.

```
			// This blocks until server exits
			err := m.nfsServer.Run(m.context)
			if err != nil {
				m.logger.WithError(err).Error("nfs server exited with error")
			}
```

In fact, the only way the loop is repeatedly executed is if we fail waiting for the attach. All other paths out of default lead to shutdown.

```
		default:
			if !volume.CheckDeviceValid(devicePath) {
				m.logger.Warn("Waiting with nfs server start, volume is not attached")
				break
			}
```

Reply:
Now that I understand the flow, do you think you could add a comment before the loop (not your code) as part of this PR so I don't make the same mistake again?

Something like the following (or whatever you think helps make it clear).

```go
        // Check every waitBetweenChecks for volume attachment. Then run server once to completion.
	for ; ; time.Sleep(waitBetweenChecks) {
```